packages/rocketchat-api/server/v1/users.js

```diff
@@ -260,13 +260,48 @@ RocketChat.API.v1.addRoute('users.update', { authRequired: true }, {
 	}
 });
 
+RocketChat.API.v1.addRoute('users.updateOwnBasicInfo', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			userId: String,
+			data: Match.ObjectIncluding({
+				email: Match.Maybe(String),
+				name: Match.Maybe(String),
+				username: Match.Maybe(String),
+				actualPassword: Match.Maybe(String),
+				newPassword: Match.Maybe(String)
+			}),
+			customFields: Match.Maybe(Object)
+		});
+
+		const userData = {
+			_id: this.bodyParams.userId,
+			email: this.bodyParams.data.email,
+			realname: this.bodyParams.data.name,
+			username: this.bodyParams.data.username,
+			newPassword: this.bodyParams.data.newPassword
+		};
+
+		if (this.bodyParams.data.actualPassword) {
+			userData.typedPassword = SHA256(this.bodyParams.data.actualPassword);
+		}
+
+		Meteor.runAsUser(this.bodyParams.userId, () => Meteor.call('saveUserProfile', userData, this.bodyParams.customFields));
```

Comment:
That way you are allowing anyone to change other's user information, should run as the authenticated user